api/transport/request.go

```diff
@@ -31,6 +31,23 @@ import (
 
 // Request is the low level request representation.
 type Request struct {
+	// ID is a unique identifier for a request/response pair. This MAY be a
```

Comment:
IMO from the point of view of the server, there doesn't have to be a
uniqueness guarantee. In fact, it may be a bad idea to imply that there is a
uniqueness guarantee because servers shouldn't rely on, say,
`map[ID]something`.

The server should only follow this contract: the ID in the returned response
must be the same as the ID in the corresponding request.

Because clients choose the ID when they send the request, they can ascribe any
meaning to it, including client-scoped uniqueness.

Reply:
sounds reasonable to me